packages/app-project/src/screens/ProjectHomePage/components/Hero/HeroContainer.spec.js

```diff
@@ -0,0 +1,23 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+
+import HeroContainer from './HeroContainer'
+import Hero from './Hero'
+
+let wrapper
+let componentWrapper
+
+describe('Component > HeroContainer', function () {
+  before(function () {
+    wrapper = shallow(<HeroContainer.wrappedComponent />)
+    componentWrapper = wrapper.find(Hero)
+  })
+
+  it('should render without crashing', function () {
+    expect(wrapper).to.be.ok()
+  })
+
+  it('should render the `Hero` component', function () {
+    expect(componentWrapper).to.have.lengthOf(1)
+  })
+})
```

Comment:
Could we have some tests for the `componentDidMount` async request and the expected component states: success and error?

Reply:
Sure - are there any examples of testing async component methods kicking around?

Reply:
I don't think we any other components directly making async HTTP calls in their component lifecycle methods, but I think we can take a similar approach as what we do in the store tests and leverage mocha's describe blocks to make sure the setup is correct for each scenario:

1. Stub the actual network request with nock
2. Setup three describe blocks, one for each state of the request: loading, success, and error
3. For the loading scenario, you want to run the expectation synchronously, so instantiate the wrapper instance in the same `it` statement function and run the expectation to check the state
4. For the success and error scenarios, we want to be sure the async call is finished before calling the expectation, so you'll want to instantiate the wrapper in a before or beforeEach function. Then the expectations check the component state is what we expect.

Since I've done so much with testing lately, I'd be happy together a rough example of what I'm describe in a branch off of this one if you'd like to have a fuller explanation of what I'm thinking.

Reply:
That'd be super helpful, actually, I'm just about to get stuck into PRN stuff. I think this is the last thing to fix in this PR aside from the layout stuff I was going to tackle with @artsy/fresnel - mind if I do that in a separate PR?

Reply:
https://github.com/zooniverse/front-end-monorepo/pull/1116

Reply:
the approach i settled on was waiting for the target method to resolve by spying on it, then testing the child component props for what i was expecting

Reply:
I've also read that you can return a promise from a `before` or `beforeEach` block, in which case mocha will wait on running the tests until those promises resolve.